File: Stack/JavaScript/Stack.js
'use strict';

class Stack {

    constructor() {
        this.stack = [];
        this.top = 0;
    }

    push(item) {
        this.stack[this.top++] = item;
    }

    length() {
        return this.top;
    }

    peek() {
        return this.stack[this.top - 1];
    }

    isEmpty() {
        return this.top === 0;
    }

    pop() {

        if (this.isEmpty()) {
            console.log('The Stack is empty. It is impossible to remove element.');
            return;
        }

        --this.top;
        return this.stack.pop();
    }

    print() {

        let top = this.top - 1;

        while (top >= 0) {
            console.log(this.stack[top]);
            top--;
        }

    }

}

File: Stack/Python/stack.py
from typing import List

class Stack:

    def __init__(self, capacity: int):
        if capacity <= 0:
            raise Exception('Capacity should be greater than 0.') 

        self.stack: List[int] = []
        self.capacity: int = capacity
        self.top: int = 0

    def size(self) -> int:
        return self.top

    def isEmpty(self) -> bool:
        return self.top == 0   

    def isFull(self) -> bool:
        return self.top >= self.capacity

    def push(self, item: int):
        if(self.isFull()):
            raise Exception('The Stack is overflow. It is impossible to add new element.') 

        self.stack += [item]
        self.top += 1

    def pop(self) -> int:
        if(self.isEmpty()):
            raise Exception('The Stack is empty. It is impossible to remove element.')

        self.top -= 1
        item = self.stack[self.top]
        del(self.stack[self.top])
        return item

    def peek(self) -> int:
        if(self.isEmpty()):
            raise Exception('The Stack is empty. It is impossible to peek element.')  

        return self.stack[self.top - 1]   

    def print(self):
        for i, item in enumerate(self.stack):
            print(i, item)     
